src/wallet/wallet.h

```diff
@@ -927,7 +945,8 @@ class SpendableInputs {
     bool limited = false;
 
 public:
-    std::vector<COutput> utxos;
+    /// Carry the UA associated with each COutput to track linking of account addresses.
+    std::vector<std::pair<COutput, std::optional<AddressUFVKMetadata>>> utxos;
```

Comment:
I _think_ it needs the diversifier, because a UFVKId on its own only tells you which account its associated with, not which UA, right? In https://github.com/zcash/zcash/pull/6377/files/5c0beb3af9ab7949197a0d847978f16718457653..7859f55665c3e8b286ea8faebb1826d8b8aa96c8#diff-7bc4ee2e089be48fc4e40c43202e50ff4297c1d5ce4c47607aea01d128967aa0R504-R508 we check that we have only a single distinct `AddressUFVKMetadata` to ensure we’re not linking account addresses.

However, as I mention in https://github.com/zcash/zcash/pull/6377/files/5c0beb3af9ab7949197a0d847978f16718457653..7859f55665c3e8b286ea8faebb1826d8b8aa96c8#r1128690816, since we may be changing the semantics of this soon enough, it might be better to just not have `GetRequiredPrivacyPolicy` check for `ALAA` at the moment, and then introduce the simpler (breaking) semantics in a separate PR.

Reply:
Hm, ok. The other change that @str4d and I had discussed in relation to this was getting rid of the optionality of the diversifier index in `AddressUFVKMetadata` - which I guess we can do later. The thing is, if we do the strengthening of `AllowRevealedSenders` and `ALAA` in this PR, then we don't need to introduce this pair at all, since we'll require `ALAA` if more than one taddr is represented in the set of outputs.

The way I see it, it's kind of an either/or question, more than a this-then-that. We can leave this data structure as-is if we strengthen those policies in this PR; otherwise, we will introduce this change now only to more or less immediately revert it.

Reply:
Okay, I get what you're talking about now. I'm :+1: to leaving the ALAA checks out of `GetRequiredPrivacyPolicy` for this PR if we can then do the strengthening immediately after.

Reply:
Are we official on the strengthening? (@str4d)

I guess we don’t have docs for privacy policies yet (#6240, #6418, #6455), so we’re pretty free to change things 😁

Reply:
Also, I’m certain I’ll have the strengthening PR up before _this_ PR is merged – I already had it implemented that way once (after our pairing a week or two ago).

Reply:
Hm, would it make sense to do the `TransparentCoinbasePolicy` bit along with that, rather than doing it in this PR?

Reply:
I don’t think so.
1. `TransparentCoinbasePolicy` is one of the things that made me ok with rolling back the not-good “account proxy” change (https://github.com/zcash/zcash/pull/6377#discussion_r1128838809),
2. `TCP` replaces three other approaches to allow/require coinbase, one of which was introduced in this PR, so postponing that would re-introduce those issues, and
3. I think it’s a bit more baked in to this PR than the `ALAA` changes.

Reply:
See #6467.